kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/index.vue

```diff
@@ -174,16 +174,16 @@
                 },
               });
             } else {
-              if (this.$route.params.quizId === exam.id) {
-                return;
-              }
               this.$router.replace({
                 name: PageNames.EXAM_CREATION_ROOT,
                 params: {
                   classId: this.$route.params.classId,
                   quizId: exam.id,
                 },
               });
+              // Update the quiz with the new id - there won't be if this is a new quiz,
+              // this ensures the quiz is kept up-to-date since we're replacing the route
+              this.updateQuiz({ id: exam.id });
```

Comment:
This seems right - although maybe this should live inside the useQuizCreation composable?

Reply:
I tried putting it at the end of saveQuiz but it wasn't working so I tried this... I think that maybe the issue was that I wasn't returning the exam in the `then` callback on the `ExamResource.saveModel` -- I've pushed an update that seems to work better there.